samcli/lib/build/build_graph.py

```diff
@@ -625,7 +626,10 @@ def get_build_dir(self, artifact_root_dir: str) -> str:
         Return the directory path relative to root build directory
         """
         self._validate_functions()
-        return self.functions[0].get_build_dir(artifact_root_dir)
+        build_dir = self.functions[0].get_build_dir(artifact_root_dir)
+        if is_experimental_enabled(ExperimentalFlag.BuildImprovements22) and len(self.functions) > 1:
+            build_dir = f"{build_dir}-Shared"
```

Comment:
How is this postfix "Shared" determined?

Reply:
When determining the artifacts directory that we will have our resultant build in, before we would copy all the contents in the same code URI (technically the same build definition) over and over. Now if there are multiple functions with the same build definition we will just put them into one single shared artifacts directory.

Reply:
I see, putting up that comment above the line and also making the suffix a constant string thats defined and importable would be good.